e2e-tests/src/waiting.rs

```diff
@@ -1,31 +1,6 @@
 use std::sync::mpsc::channel;
-
-use codec::Decode;
-use log::{error, info};
-use substrate_api_client::ApiResult;
-
-use crate::{Connection, Header};
-
-pub fn wait_for_event<E: Decode + Clone, P: Fn(E) -> bool>(
-    connection: &Connection,
-    event: (&str, &str),
-    predicate: P,
-) -> anyhow::Result<E> {
-    let (module, variant) = event;
-    info!("[+] Creating event subscription {}/{}", module, variant);
-
-    let (events_in, events_out) = channel();
-    connection.subscribe_events(events_in)?;
-
-    loop {
-        let args: ApiResult<E> = connection.wait_for_event(module, variant, None, &events_out);
-        match args {
-            Ok(event) if predicate(event.clone()) => return Ok(event),
-            Ok(_) => (),
-            Err(why) => error!("Error {:?}", why),
-        }
-    }
-}
+use log::info;
+use common::{Connection, Header};
 
 /// blocks the main thread waiting for a block with a number at least `block_number`
 pub fn wait_for_finalized_block(connection: &Connection, block_number: u32) -> anyhow::Result<u32> {
```

Comment:
Should this function be moved to `common` as well? Or we just doing part of refactoring in this PR only. I'm just thinking out loud what should be in `common` anyway.

Reply:
I've been trying to make this refactor pretty minimal, since I'm not sure what actually should be in the common crate and I didn't want to spend too much time on this. So I'm afraid this stays here.